pkg/wharfyml/parse.go

```diff
@@ -63,9 +71,20 @@ func decodeFirstDoc(reader io.Reader) (*yaml.Node, error) {
 		// Continue, but only parse the first doc
 		return body, ErrTooManyDocs
 	}
+	body = unwrapNodeRec(body)
 	return body, nil
 }
 
+func unwrapNodeRec(node *yaml.Node) *yaml.Node {
+	for node.Alias != nil {
+		node = node.Alias
+	}
+	for i, child := range node.Content {
+		node.Content[i] = unwrapNodeRec(child)
```

Comment:
`node` is received as a pointer and `Node.Content` is a slice of pointers, so this assignment shouldn't be necessary.
More future-proof and less prone to bad stuff to keep it in though, just wanted to bring it up in case you hadn't considered it. Basically no performance gain I'd assume, maybe it gets optimized away anyways 🤷🏻

Reply:
The first for loop isn't honoring the pointer, so if only relying on pointers and any of the content nodes is an alias node then it will cause glitches.

Reply:
Oh yeah, nice catch 👍🏻